public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -202,23 +208,34 @@ export const GeneratePopoverBody: React.FC<{
   const renderInnerFooter = () => {
     return (
       <EuiPopoverFooter className="incontextInsightGeneratePopoverFooter" paddingSize="none">
-        <EuiFlexGroup gutterSize="none" direction={'rowReverse'}>
-          {insightAvailable && (
-            <EuiFlexItem
-              grow={false}
-              onClick={() => {
+        {
+          <div style={{ display: showInsight ? 'none' : 'block' }}>
+            <MessageActions
+              contentToCopy={summary}
+              showFeedback
+              showTraceIcon={insightAvailable}
```

Comment:
It seems weird to pass insight related status to Trace Icon and its related action function. 

How about add a new parameter for the purpose of extending the MessageActions with customize buttons configuration?  For this case, insight button should be passed in as a extending actions. You can think about which actions should be internal ones, and which should be extending ones.

Reply:
It seems like that the only difference between insight and trace is just the click function, so I think adding a new button is a little redundent.

Reply:
Not exactly. insight button needs to show info "Insight With RAG" when moving your cursor over it. And what if you wants to have both trace button and insight button. 

It could be an enhancement since this PR has been merged.